libUtils/windowsChromeDriver.py

```diff
@@ -2,45 +2,41 @@
 import requests
 import os
 import zipfile
+import platform
 
 class BrowserManager:
-    def __init__(self,extract_dir=".//dependencies//chrome-driver"):
-        self.extract_dir = extract_dir
-    def get_chrome_version(self):   #    Get the Chrome browser version installed on Windows.
+
+    def __init__(self):
+        self.driver_path = ".//dependencies//chrome-driver"
+        # self.base_url = f"https://storage.googleapis.com/chrome-for-testing-public/{chrome_version}/win64/chromedriver-win64.zip"
+        self.reg_path = r'reg query "HKEY_CURRENT_USER\Software\Google\Chrome\BLBeacon" /v version'
+
+    def get_chrome_version(self):   #Get the Chrome browser version installed on Windows.
         try:
-            output = subprocess.check_output(
-                r'reg query "HKEY_CURRENT_USER\Software\Google\Chrome\BLBeacon" /v version',
-                shell=True,
-                encoding='utf-8'
-            )
+            output = subprocess.check_output(self.reg_path, shell=True, encoding='utf-8')
             # Extract version from the command output
             version = output.split()[-1]
             print(f"Chrome Version: {version}")
             return version
-
         except Exception as e:
-            raise Exception(f"Error getting Chrome version: {e}")
+            print( f"Failed to read chrome browser version from path: {self.reg_path}")
+            raise Exception(f"Failed to get chrome browser version. Error: {e}")
 
-    def get_chrome_driver_download_url(self, chrome_version):
+    def get_driver_url(self,chrome_version):
         try:
-            if not chrome_version:
-                raise Exception("Chrome version is required to fetch the download URL.")
-
-            download_url = f"https://storage.googleapis.com/chrome-for-testing-public/{chrome_version}/win64/chromedriver-win64.zip"
-            print(f"Download URL: {download_url}")
+            architecture = 'win64' if platform.architecture()[0] == '64bit' else 'win32'
+            print(architecture)
+            download_url = f"https://storage.googleapis.com/chrome-for-testing-public/{chrome_version}/{architecture}/chromedriver-{architecture}.zip"
+            print(f"Chrome Driver Download URL: {download_url}")
 
             response = requests.head(download_url)
-
             if response.status_code != 200:
-                raise Exception(
-                    f"Failed to get ChromeDriver for Chrome {chrome_version}. HTTP Status Code: {response.status_code}")
-
+                raise Exception(f"Failed to get ChromeDriver for Chrome {chrome_version}.")
```

Comment:
Requirement not met: "Add a print stament before raise and print status code and response HTTP Status Code: {response.status_code}"